lib/lightning/projects/provisioner.ex

```diff
@@ -86,22 +86,26 @@ defmodule Lightning.Projects.Provisioner do
     |> Enum.reduce(
       Multi.new(),
       fn workflow_changeset, multi ->
-        additional_multi =
-          classify_audit(workflow_changeset)
-          |> case do
-            {:no_action, _nil} ->
-              Multi.new()
-
-            {action, workflow_id} ->
-              audit_workflow_multi(action, workflow_id, user_or_repo_connection)
-          end
-
-        Multi.append(multi, additional_multi)
+        append_audit_multi(workflow_changeset, multi, user_or_repo_connection)
       end
     )
     |> Repo.transaction()
   end
 
+  defp append_audit_multi(workflow_changeset, multi, user_or_repo_connection) do
+    additional_multi =
+      classify_audit(workflow_changeset)
+      |> case do
+        {:no_action, _nil} ->
+          Multi.new()
+
+        {action, workflow_id} ->
+          audit_workflow_multi(action, workflow_id, user_or_repo_connection)
+      end
+
+    Multi.append(multi, additional_multi)
```

Comment:
Sorry for being pick again Rory, one small refactoring to simplify it a bit more:
```
case classify_audit(workflow_changeset) do
        {:no_action, _nil} ->
          multi

        {action, workflow_id} ->
          audit_multi = audit_workflow_multi(action, workflow_id, user_or_repo_connection) 
          Multi.append(multi, audit_multi)
      end
```